core/Stream/pfSizedStream.cpp

```diff
@@ -41,37 +41,43 @@ void pfSizedStream::seek(uint32_t pos)
     }
 
     fBase->seek(fBegin + pos);
+    fPos = pos > fLength ? fLength : pos;
```

Comment:
The check above is just a warning, so I feel like we should just set `fPos = pos` directly here, even though that puts it out of range.  More importantly, doing so more closely matches the behavior of the code before these changes.